src/components/schedule-visualizer/door.tsx

```diff
@@ -0,0 +1,139 @@
+import { makeStyles } from '@material-ui/core';
+import * as RomiCore from '@osrf/romi-js-core-interfaces';
+import React from 'react';
+import { Spinner } from './spinner';
+
+// Door state
+export enum DoorStates {
+  CLOSE,
+  PROCESS,
+  OPEN,
+}
+
+// Door Type
+const DOOR_TYPE_SINGLE_SLIDING = 1;
+const DOOR_TYPE_DOUBLE_SLIDING = 2;
+export interface DoorProps {
+  door: RomiCore.Door;
+  onClick?(e: React.MouseEvent<SVGGElement>, door: RomiCore.Door): void;
+  currentMode?: number;
```

Comment:
No, in this case `currentMode` it's ok. `DoorState` is a bad name.
Now the enum it's called 
``` typescript
export enum DoorMode {
  CLOSE,
  PROCESS,
  OPEN,
}
```
Because the `State` contains the `currentMode` of the door